### PR TITLE
BoxHelper: More consistent API

### PR DESCRIPTION
**Pull request by Mugen87**
It's more consistent to other helpers if the `.update()` method has no arguments.

See https://github.com/mrdoob/three.js/issues/7824#issuecomment-166152863

### CODE REVIEW

File: src/helpers/BoxHelper.js [hunk 1]
```diff
@@ -38,15 +39,15 @@ BoxHelper.prototype.update = ( function () {
...
+		if ( object !== undefined ) {
 
-			box.copy( object );
+			console.warn( 'THREE.BoxHelper: .update() has no longer arguments.' );
 
-		} else {
+		}
 
-			box.setFromObject( object );
+		this.object.updateMatrixWorld( true );
```
Comment (WestLangley) [inline src/helpers/BoxHelper.js]: Why did you add this here?

Comment (Mugen87) [inline src/helpers/BoxHelper.js]: I've just realize that `Box3 .expandByObject()` calls `.updateMatrixWorld()`. I'll remove this line!

File: src/helpers/BoxHelper.js [hunk 2]
```diff
@@ -83,7 +84,7 @@ BoxHelper.prototype.update = ( function () {
 
 		position.needsUpdate = true;
 
-		this.geometry.computeBoundingSphere();
```
Comment (WestLangley) [inline src/helpers/BoxHelper.js]: I expect it is wise to update the bounding sphere whenever a geometry's vertices are modified. Whatever we do, it should be consistent.

Comment (Mugen87) [inline src/helpers/BoxHelper.js]: Okay, was not sure about that. I'll keep this in mind!

File: src/helpers/BoxHelper.js [hunk 3]
```diff
@@ -83,7 +84,7 @@ BoxHelper.prototype.update = ( function () {
...
+		return this;
```
Comment (WestLangley) [inline src/helpers/BoxHelper.js]: It is not clear to me the usefulness of returning `this` from the `update()` method. Again, whatever we do, it should be consistent.

Comment (Mugen87) [inline src/helpers/BoxHelper.js]: Oh, i've seen this in `VertexNormalsHelper`.

Comment (Mugen87) [inline src/helpers/BoxHelper.js]: Maybe correct `VertexNormalsHelper .update()` with a separate PR?

Comment (WestLangley) [inline src/helpers/BoxHelper.js]: I fixed `VertexNormalsHelper` and `FaceNormalsHelper` in the dev branch.

**Comment by Mugen87** [comment]
Okay, i think this is ready to merge.

@WestLangley Thank you for your kind quality assurance :blush:

**Comment by mrdoob** [comment]
I think the biggest user of this function is the editor:

```js
https://github.com/mrdoob/three.js/blob/dev/editor/js/Viewport.js#L63
```

Maybe we want to add a `setFromObject()` function?

**Comment by WestLangley** [comment]
>Maybe we want to add a setFromObject() function?

It would just implement
```
this.object = object;
this.update();
```

I wonder if that is necessary, given `BoxHelper.object` is exposed? It is up to you.

**Comment by mrdoob** [comment]
It'll follow `Box3` a bit...

https://github.com/mrdoob/three.js/blob/bf75413322057d8803afac240a59abc7d94f1d20/src/math/Box3.js#L126

**Comment by Mugen87** [comment]
Sry, i've missed the usage in the editor :disappointed:

So the new method could look like this then, correct?
```javascript
BoxHelper.prototype.setFromObject = function ( object ) {

   this.object = object;
   this.update();

}
```

**Comment by WestLangley** [comment]
Yes.

**Comment by Mugen87** [comment]
@WestLangley I've added a `return this;` to `.setFromObject()` because equivalent methods also do this. Hope this it's okay like this.

**Comment by mrdoob** [comment]
I'm thinking that we could just rename `update()` to `setFromObject()` and keep the same functionality.

**Comment by WestLangley** [comment]
We need an `update()` method that can be called inside the render loop -- just like we do with `VertexNormalsHelper`, for example.

**Comment by Mugen87** [comment]
Besides, the central motivation of this PR is to unify the `.update()` method. Users might ask: _Why is `BoxHelper .update()` the only update method that requires a parameter?_ Having a consistent method signature helps to ensure consistency.

**Comment by mrdoob** [comment]
I see I see... I guess I'm not happy with my original design 😅

**Comment by mrdoob** [comment]
Thanks guys!

**Comment by WestLangley** [comment]
@mrdoob How would you prefer it? Maybe we can change the API.

**Comment by mrdoob** [comment]
@WestLangley I'll think about it.

**Comment by aardgoose** [comment]
note: before the change BoxHelper would accept a Box3() object as a constructor parameter as well as an instance of Object3D.  This change should probably be documented in the "release notes", or a warning emitted, if it is intentional?

( I was using this behaviour, but not a big problem for me at least)

**Comment by Mugen87** [comment]
I'll note this in the migration guide: https://github.com/mrdoob/three.js/wiki/Migration-Guide
